artemis/reporting/export.py

```diff
@@ -0,0 +1,305 @@
+import collections
+import copy
+import dataclasses
+import datetime
+import json
+import os
+import urllib.parse
+from collections import Counter, defaultdict
+from pathlib import Path
+from typing import Any, DefaultDict, Dict, List, Set
+
+import tqdm
+import typer
+from jinja2 import BaseLoader, Environment, StrictUndefined
+
+from artemis.config import Config
+from artemis.db import DB
+from artemis.json_utils import JSONEncoderAdditionalTypes
+from artemis.utils import is_ip_address
+
+from .base.language import Language
+from .base.report import Report
+from .base.report_type import ReportType
+from .base.reporters import reports_from_task_result
+from .base.templating import build_message_template
+from .utils import get_top_level_target_if_present
+
+environment = Environment(loader=BaseLoader(), undefined=StrictUndefined, trim_blocks=True, lstrip_blocks=True)
+OUTPUT_LOCATION = "./output/autoreporter/"
+
+
+def _all_reports_are_old(reports: List[Report], threshold_days: int = 60) -> bool:
+    for report in reports:
+        if report.timestamp and report.timestamp >= datetime.datetime.now() - datetime.timedelta(days=threshold_days):
+            return False
+    return True
+
+
+def _build_subsequent_reminder(report: Report) -> Report:
+    new_report = copy.deepcopy(report)
+    new_report.is_subsequent_reminder = True
+    return new_report
+
+
+def _get_domain_or_ip_only(data: str) -> str:
+    """Extracts domain/ip from URL or host:port string."""
+    if "://" in data:
+        host = urllib.parse.urlparse(data).hostname
+        assert host
+        return host
+    if ":" in data:
+        host = urllib.parse.urlparse("//" + data).hostname
+        assert host
+        return host
+    return data
```

Comment:
This shouldn't assert, but raise `ValueError` or smth like that. I'm not a fan of this function either way, but it does seem to work.